snap/snapcraft.yaml

```diff
@@ -178,19 +180,22 @@ apps:
     daemon: oneshot
     start-timeout: 15m
     plugs: [network]
-  # This is a simple service which calls into vault to retrieve the Redis password and then makes a
-  # Redis call to set the password. As such, Redis is now started after security-secretstore-setup
-  # has run, following by this service. Once the password has been set, this service exits. In the
-  # Docker version, the entrypoint.sh waits until it is interrupted before it exits.
+  # This is a simple service which calls into vault to retrieve the Redis password and then 
+  # generate Redis config file for Redis to start up with credentials and ACL rules. 
+  # Redis should be start once the doneFile is created. Once the config file has been generated and 
+  # verified authenticated connection, this service exits. In the Docker version, 
+  # the customized redis' entrypoint.sh performs the similar actions as described above.
```

Comment:
Might it not be more efficient for the snap to create a wrapper script for launching redis rather than creating a separate service?

Reply:
For that matter, the snap install could write the config file and the redis sevice could just start redis directly pointing at it.

Reply:
Right, the key or major purpose of that bootstrap-redis is to "dynamically" generate the config file as the credentials are generated from `secretstore-setup` during the run time for the first time. Hence, you can NOT just do the snap install.

Reply:
@tonyespy Maybe a command-chain would be useful here?

Reply:
Jim is just converting an existing oneshot service in the snap which used to call security-bootstrap-redis. The redis service is already defined to start after this service, so this transition to using bootstrap doesn't really impact the service ordering.

Yes, we could re-implement this via command-chain, however I don't see any added benefit to doing so, unless I'm missing something?